### PR TITLE
Change to notify stopped only if less than watched percent

### PR DESCRIPTION
**Pull request by JonnyWong16**
- Also fix paused notifications sending at the end of items (right before stopping)

### CODE REVIEW

File: plexpy/activity_handler.py [hunk 1]
```diff
@@ -187,11 +194,11 @@ def process(self):
                                              view_offset=self.timeline['viewOffset'])
                     # Start our state checks
                     if this_state != last_state:
-                        if this_state == 'paused':
+                        if this_state == 'paused' and progress_percent < 99:
```
Comment (drzoidberg33) [inline plexpy/activity_handler.py]: You shouldn't change it here, these `on_` functions do more than notifications. Especially the `stopped` state which triggers all the processing.

File: plexpy/activity_handler.py [hunk 2]
```diff
@@ -83,8 +85,10 @@ def on_stop(self, force_stop=False):
             db_session = ap.get_session_by_key(session_key=self.get_session_key())
 
             # Fire off notifications
-            threading.Thread(target=notification_handler.notify,
-                             kwargs=dict(stream_data=db_session, notify_action='stop')).start()
+            progress_percent = helpers.get_percent(self.timeline['viewOffset'], db_session['duration'])
+            if progress_percent < plexpy.CONFIG.NOTIFY_WATCHED_PERCENT:
```
Comment (drzoidberg33) [inline plexpy/activity_handler.py]: What if we're not using watched notifications and rely on stopped notifications or if we want both?

These notification changes should be something we can toggle - one setting toggle is fine with me. Something like `permit consecutive notifications` which you can disable to achieve this.

File: plexpy/activity_handler.py [hunk 3]
```diff
@@ -66,6 +66,8 @@ def on_start(self):
 
     def on_stop(self, force_stop=False):
         if self.is_valid_session():
+            from plexpy import helpers
```
Comment (Hellowlol) [inline plexpy/activity_handler.py]: Whats up with all the inline imports?

Comment (JonnyWong16) [inline plexpy/activity_handler.py]: Copy and paste. Fixed.

**Comment by drzoidberg33** [comment]
Regarding the notification stuff - I don't know if this is the right way to do it.

We are logging all the timestamps for notifications in the `notify_log` - I would write some getter/setters for these so we have easy access to retrieving them then use these values to determine whether notifications should fire off. All the logic should be done in the `notification_handler` - that code is a mess I know :smile:

**Comment by JonnyWong16** [comment]
There's not much in `notify_log` to use...Time stamps of when a notification was sent aren't really the best for writing logic, besides maybe something like `on_stop > on_watched`.

What about just putting the logic right before "Build and send notification" in `notification_handler`? And still set the notification in `notify_log`.

**Pull request closed without merging**